src/VisualStudio/Core/Def/Implementation/EditAndContinue/VsENCRebuildableProjectImpl.cs

```diff
@@ -590,10 +590,13 @@ private void AddActiveStatements(Solution solution, ShellInterop.ENC_ACTIVE_STAT
 
                 var flags = (ActiveStatementFlags)vsActiveStatement.ASINFO;
 
-                IVisualStudioHostDocument vsDocument = _vsProject.GetCurrentDocumentFromPath(vsActiveStatement.filename);
-                if (vsDocument != null)
+                // Finds a document id in the solution with the specified file path.
+                DocumentId documentId = solution.GetDocumentIdsWithFilePath(vsActiveStatement.filename)
+                    .Where(dId => dId.ProjectId == _vsProject.Id).FirstOrDefault();
```

Comment:
There should be zero or one, no?  Depending on IDE conventions, consider using `SingleOrDefault`.